portal-impl/src/custom-sql/portal.xml

```diff
@@ -1056,6 +1056,29 @@
 				[$TYPE$]
 		]]>
 	</sql>
+	<sql id="com.liferay.portal.kernel.service.persistence.RoleFinder.countU_GByRoleId">
+		<![CDATA[
+			SELECT SUM(TEMP_TABLE) AS COUNT_VALUE
```

Comment:
Using a temporary table seems like a good solution to this, but since it's a new pattern you may want to check with Brian or someone else with more SQL knowledge than myself to figure out if it's the best solution in terms of performance, compatibility, etc.